mitmproxy/proxy/protocol/http.py

```diff
@@ -407,6 +407,8 @@ def get_response():
                 self.channel.ask("responseheaders", f)
 
             self.log("response", "debug", [repr(f.response)])
+            if hasattr(self, 'has_tailers') and self.has_tailers:
```

Comment:
Is `has_tailers` necessary? Shouldn't `flow.trailers == None` be the same information?
If we really need `has_tailers`, please make it a first-class citizen of the Response object (instead of checking if it has been set during runtime), so it also serializes properly, and we can remove the `hasattr` checks.

Reply:
When process  HTTP/1.1 flow it will throw 
>AttributeError: 'RootContext' object has no attribute 'has_tailers'  

I can't figure out how to update `has_tailers` in `Http2SingleStreamLayer` if it is a attribute of Response object

Reply:
Do you still have this problem, now that you moved the trailers to the response class?